src/util/records.py

```diff
@@ -1147,6 +1148,8 @@ def add_ref(ref):
             doc = lxml.etree.parse(fp)
             for node in doc.xpath(xpath):
                 found = True
+                if xml_filename is None and module == from_module:
```

Comment:
Why do we check module vs from_module? Don't we need to set the xml_filename to be the actual file always?

Reply:
I think we should do what standard code does.